libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -497,6 +497,11 @@ private void initializeView() {
     inflate(getContext(), R.layout.navigation_view_layout, this);
     bind();
     initializeNavigationViewModel();
+    initializeNavigationRouter();
+    initializeNavigationLocationEngine();
+    initializeNavigationUtils();
+    initializeNavigationConnectivityController();
+    initializeNavigationLocationEngine();
```

Comment:
As these don't require `NavigationViewOptions` and can be created at the time of retrieval of the `ViewModel`, maybe it makes sense to use a `ViewModelFactory`?